### PR TITLE
Add Community Management Interface and SIG Management Interface

### PR DESCRIPTION
**Pull request by lishengbao**

### CODE REVIEW

File: app/graphql/types/queries/community_detail_overview_query.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Types
+  module Queries
+    class CommunityDetailOverviewQuery < BaseQuery
+
+      attr_accessor :indexer, :repo_urls, :begin_date, :end_date, :origin
```
Comment (EdmondFrank) [inline app/graphql/types/queries/community_detail_overview_query.rb:7]: Why use attr_accessor and instance variables on this query ? @lishengbao

File: app/graphql/types/queries/contribution_detail_overview_query.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      argument :begin_date, GraphQL::Types::ISO8601DateTime, required: false, description: 'begin date'
+      argument :end_date, GraphQL::Types::ISO8601DateTime, required: false, description: 'end date'
+
+      def resolve(label: nil, level: 'repo', begin_date: nil, end_date: nil)
+        label = ShortenedLabel.normalize_label(label)
+
+        login_required!(context[:current_user])
+        validate_by_label!(context[:current_user], label)
+
+        @begin_date, @end_date, interval = extract_date(begin_date, end_date)
```
Comment (EdmondFrank) [inline app/graphql/types/queries/contribution_detail_overview_query.rb:20]: Same question as above.

File: app/graphql/types/queries/user_search_query.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        login_required!(context[:current_user])
+
+        User.joins(:login_binds)
+            .where("users.email LIKE :search
+                    OR users.name LIKE :search
+                    OR login_binds.account LIKE :search
+                    OR login_binds.nickname LIKE :search", search: "%#{keyword}%")
+end
+    end
+  end
```
Comment (EdmondFrank) [inline app/graphql/types/queries/user_search_query.rb:20]: code format

File: app/models/commit_feedback.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      }
+    }
+  end
+
+  def self.fetch_commit_feedback_list(repo_urls, values, target: 'repo_name.keyword', value_field: 'id.keyword', state: nil)
+    base = self.must(terms: { target => repo_urls })
+               .must(terms: { value_field => values })
+    if state.nil?
+      base = base.where( 'state.keyword': state)
+    end
```
Comment (EdmondFrank) [inline app/models/commit_feedback.rb:45]: Opposite logic?

File: app/models/commit_feedback.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+               .execute
+               .raw_response
+    (resp&.[]('hits')&.[]('hits') || []).map do |data|
+      data["_source"]
+    end
+  end
+
+  def self.fetch_commit_feedback_one(repo_urls, values, target: 'repo_name.keyword', value_field: 'id.keyword', state: nil)
+    fetch_list = fetch_commit_feedback_list(repo_urls, values, target: target, value_field: value_field, state: state)
+    fetch_list.first
```
Comment (EdmondFrank) [inline app/models/commit_feedback.rb:56]: No need for intermediate variables: `fetch_list`

File: app/models/concerns/commit_enrich.rb [hunk 6]
```diff
@@ -130,31 +136,38 @@ def merge_commit_organization(source, target)
...
+                 .execute
+                 .raw_response
+      (resp&.[]('hits')&.[]('hits') || []).map do |data|
+        data["_source"]
+      end
+    end
+
+    def fetch_commit_one_by_hash(repo_urls, hash_value, target='tag')
+      fetch_list = fetch_commit_list_by_hash(repo_urls, [hash_value], target)
+      fetch_list.first
```
Comment (EdmondFrank) [inline app/models/concerns/commit_enrich.rb:169]: same as above

File: app/models/concerns/pull2_enrich.rb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+
+    def fetch_check_agg_list_by_repo_urls(repo_urls, begin_date, end_date, target: 'tag',
+                          filter: :comment_created_at, sort: :comment_created_at, direction: :asc, filter_opts: [], sort_opts: [],
+                          user_login_list: [])
+      base = base_terms_by_repo_urls(repo_urls, begin_date, end_date, target: target,
+                              filter: filter, sort: sort, direction: direction, filter_opts: filter_opts, sort_opts: sort_opts)
+      if user_login_list.present?
+        base = base.where( user_login: user_login_list)
+      end
```
Comment (EdmondFrank) [inline app/models/concerns/pull2_enrich.rb:31]: code format

File: app/models/concerns/pull_enrich.rb [hunk 8]
```diff
@@ -18,17 +18,37 @@ def on_each(args)
...
-                          commit_hash_list: [])
+                          commit_hash_list: [], pr_url_list: [])
       base = base_terms_by_repo_urls(repo_urls, begin_date, end_date, target: target,
                               filter: filter, sort: sort, direction: direction, filter_opts: filter_opts, sort_opts: sort_opts)
       if commit_hash_list.present?
         base = base.where( commits_data: commit_hash_list)
       end
+      if pr_url_list.present?
+        base = base.where( url: pr_url_list)
+      end
```
Comment (EdmondFrank) [inline app/models/concerns/pull_enrich.rb:29]: format

File: app/models/concerns/pull_enrich.rb [hunk 9]
```diff
@@ -18,17 +18,37 @@ def on_each(args)
...
+                 .execute
+                 .raw_response
+      (resp&.[]('hits')&.[]('hits') || []).map do |data|
+        data["_source"]
+      end
+    end
+
+    def fetch_pull_one_by_hash(repo_urls, hash_value, target='tag')
+      fetch_list = fetch_pull_list_by_hash(repo_urls, [hash_value], target)
+      fetch_list.first
```
Comment (EdmondFrank) [inline app/models/concerns/pull_enrich.rb:50]: same as above

File: app/graphql/mutations/create_commit_feedback.rb [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+    argument :level, String, required: false, description: 'repo or community', default_value: 'repo'
+    argument :commit_feedback_input, Input::CommitFeedbackInput, required: true, description: 'repo extension'
+
+    def resolve(label: nil, level: 'repo', commit_feedback_input: nil)
+      label = ShortenedLabel.normalize_label(label)
+      login_required!(context[:current_user])
+
+      git_indexer, repo_urls = select_idx_repos_by_lablel_and_level(label, level, GiteeGitEnrich, GithubGitEnrich)
+      commit_data = git_indexer.fetch_commit_one_by_hash(repo_urls, commit_feedback_input[:commit_hash])
+      raise GraphQL::ExecutionError.new I18n.t('commit_feedback.invalid_commit_hash') if commit_data.nil?
```
Comment (EdmondFrank) [inline app/graphql/mutations/create_commit_feedback.rb:19]: @lishengbao And I don't see a change in the i18n file, but it is used inside the code file, has the relevant file been submitted?

**Review by EdmondFrank — CHANGES_REQUESTED** [review]
Please ensure this

**Comment by lishengbao** [comment]
Done

**Comment by lishengbao** [comment]
@EdmondFrank Done

**Pull request merged**